Data/ByteString/Lazy.hs

```diff
@@ -525,6 +526,49 @@ minimum (Chunk c cs) = foldlChunks (\n c' -> n `min` S.minimum c')
                                      (S.minimum c) cs
 {-# INLINE minimum #-}
 
+-- | /O(n)/ 'compareLength' compares the length of a 'ByteString' 
+-- to an 'Int64'   
+compareLength :: ByteString -> Int64 -> Ordering
+compareLength _ toCmp | toCmp < 0 = GT
+compareLength Empty toCmp         = compare 0 toCmp
+compareLength (Chunk c cs) toCmp  = compareLength cs (toCmp - fromIntegral (S.length c))
+{-# INLINE compareLength #-}
+
+{-# RULES 
+"ByteString.Lazy compareN/length -> compareLength" [~1] forall t n.
+  compare (length t) n = compareLength t n
+  #-}
```

Comment:
How about adding a flipped version of this rule, i.e. for `compare n (length t)`?!

Reply:
Sure, but I had to implement it as `compare n (length t) = toEnum $ (2-) $ fromEnum $ compareLength t n`, which I am ashamed of. I didn't see any better way to sort of 'negate' an Ordering. If one pops to mind let me know.